src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -267,7 +267,18 @@ ExecuteDistributedModify(Task *task)
 		result = PQexec(connection, task->queryString);
 		if (PQresultStatus(result) != PGRES_COMMAND_OK)
 		{
-			ReportRemoteError(connection, result);
+			int errorCode = ReportRemoteError(connection, result);
```

Comment:
I don't really like overloading `ReportRemoteError` to return the error code.

What if we introduce a (small) helper function to answer the question "do we need to fail fast?" Maybe `ConnectionHasFailfastError`?

Then we could add a parameter `raiseError` to `ReportRemoteError` to control whether the error is logged as a `WARNING` or `ERROR`.